financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/HTTPClientUtils.java

```diff
@@ -55,89 +50,44 @@ public class HTTPClientUtils {
-
-        SSLConnectionSocketFactory sslsf = createSSLConnectionSocketFactory();
-
-        Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
-                .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
-                .register(HTTPS_PROTOCOL, sslsf)
-                .build();
-
-        final PoolingHttpClientConnectionManager connectionManager = (socketFactoryRegistry != null) ?
-                new PoolingHttpClientConnectionManager(socketFactoryRegistry) :
-                new PoolingHttpClientConnectionManager();
-
-        // configuring default maximum connections
-        connectionManager.setMaxTotal(maxTotal);
-        connectionManager.setDefaultMaxPerRoute(maxPerRoute);
+    public static CloseableHttpClient getHttpsClient() {
 
-        return HttpClients.custom().setConnectionManager(connectionManager).build();
+        return HttpsClientHolder.INSTANCE;
     }
 
-    /**
-     * Get closeable http client.
-     *
-     * @return Closeable http client
-     * @throws FinancialServicesException FinancialServicesException exception
-     */
-    @Generated(message = "Ignoring since method contains no logics")
-    public static CloseableHttpClient getHttpClient() throws FinancialServicesException {
+    private static class HttpsClientHolder {
+        private static final CloseableHttpClient INSTANCE;
 
-        Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
-                .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
-                .build();
+        static {
 
-        final PoolingHttpClientConnectionManager connectionManager = (socketFactoryRegistry != null) ?
-                new PoolingHttpClientConnectionManager(socketFactoryRegistry) :
-                new PoolingHttpClientConnectionManager();
+            INSTANCE = HttpClients.custom()
+                    .setConnectionManager(FinancialServicesCommonDataHolder.getInstance().getConnectionManager())
+                    .build();
 
-        // configuring default maximum connections
-        connectionManager.setMaxTotal(FinancialServicesConfigParser.getInstance().getConnectionPoolMaxConnections());
-        connectionManager.setDefaultMaxPerRoute(FinancialServicesConfigParser.getInstance()
-                .getConnectionPoolMaxConnectionsPerRoute());
+            // Optional: Add shutdown hook to clean up properly
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                try {
+                    INSTANCE.close();
+                } catch (IOException e) {
+                    log.error("Error shutting down HttpClient", e);
+                }
+            }));
 
-        return HttpClients.custom().setConnectionManager(connectionManager).build();
+            log.info("HTTPS Client initialized with singleton pattern.");
```

Comment:
we can make this as a debug log

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/514/commits/59d1abcd824194db81ac7185609cf05f6f591dd3